app/controllers/missions_controller.rb

```diff
@@ -66,16 +70,36 @@ def generate(mission)
     else
       flash[:error] = translate 'activerecord.errors.messages.creation_fail',
                                 model: Mission.model_name.human
+      flash[:error] << " #{mission.errors.full_messages.join(', ')}"
       redirect_to new_mission_path
     end
   end
 
   def permitted_params
+    if params[:mission][:genre] == 'regulated'
+      permit_params_of_regulated_mission
+    else
+      permit_params_of_standard_mission
+    end
+  end
+
+  def permit_params_of_regulated_mission
+    params.require(:mission).permit(
+      :name, :description, :event, :delivery_expected,
+      :recurrent, :recurrence_rule, :recurrence_end_date,
+      :max_member_count, :min_member_count,
+      :due_date, :start_date, :genre,
+      addresses_attributes: %i[id postal_code city street_name_1 street_name_2 _destroy],
+      enrollments_attributes: [:id, :_destroy, :member_id, start_time: []]
+    )
+  end
```

Comment:
Since the only difference with the standard params is only in the `enrollments_attributes` hash key, you could do something like that to simplify:

```ruby
def regulated_mission_params
  base_params.merge(enrollments_attributes: [:id, :_destroy, :member_id, start_time: []])
end

def standard_mission_params
  base_params.merge(enrollments_attributes: %i[id _destroy member_id start_time end_time])
end

def base_params
    params.require(:mission).permit(
      :name, :description, :event, :delivery_expected,
      :recurrent, :recurrence_rule, :recurrence_end_date,
      :max_member_count, :min_member_count,
      :due_date, :start_date, :genre, 
end
```
to make it more readable.

Just make sure that the merged params are properly permitted, but this is a kind of pattern that I've used before